pkg/config/handlerConfigurer.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2017 Istio Authors
```

Comment:
I don't love the super long func names; do we lose too much changing `dispatchTypesToHandlers` -> `dispatch` and `groupHandlerInstancesByTemplate` -> `groupByTmpl`?